src/lib/pubkey/pubkey.h

```diff
@@ -540,33 +540,33 @@ class BOTAN_PUBLIC_API(2, 0) PK_Decryptor_EME final : public PK_Decryptor {
 };
 
 /**
-   * Result of a key encapsulation operation.
-   */
+* Result of a key encapsulation operation.
+*/
 class KEM_Encapsulation final {
    public:
       KEM_Encapsulation(std::vector<uint8_t> encapsulated_shared_key, secure_vector<uint8_t> shared_key) :
             m_encapsulated_shared_key(std::move(encapsulated_shared_key)), m_shared_key(std::move(shared_key)) {}
 
       /**
-         * @returns the encapsulated shared secret (encrypted with the public key)
-         */
+      * @returns the encapsulated shared secret (encrypted with the public key)
+      */
       const std::vector<uint8_t>& encapsulated_shared_key() const { return m_encapsulated_shared_key; }
 
+      std::vector<uint8_t>& encapsulated_shared_key() { return m_encapsulated_shared_key; }
+
       /**
-         * @returns the plaintext shared secret
-         */
+      * @returns the plaintext shared secret
+      */
       const secure_vector<uint8_t>& shared_key() const { return m_shared_key; }
 
+      secure_vector<uint8_t>& shared_key() { return m_shared_key; }
```

Comment:
Why would we want to allow this kind of mutation?

Reply:
This seems to be driven by wanting to avoid copying the key, but IMO it's not worth it given the potential for footguns in general purpose code.

If we really wanted to support this I'd propose something akin to Rust's `fn consume(self)`:

`static std::pair<std::vector<uint8_t>, secure_vector<uint8_t>> KEM_Encapsulation::destructure(KEM_Encapsulation kem)`

which internally moves the kem and then moves the values out of the kem.